microprofile/messaging/core/src/main/java/io/helidon/microprofile/messaging/AdHocConfigBuilder.java

```diff
@@ -45,7 +45,10 @@ AdHocConfigBuilder put(String key, String value) {
     }
 
     AdHocConfigBuilder putAll(Config configToPut) {
-        configuration.putAll(configToPut.detach().asMap().orElse(Map.of()));
+        for (var e : configToPut.detach().asMap().orElse(Map.of()).entrySet()) {
```

Comment:
`orElseGet(Map::of)` is a better construct here, to avoid creating an empty map for each invocation

Reply:
(Counter: [`Map.of()` returns a `static` constant](https://github.com/openjdk/jdk/blob/jdk-21%2B35/src/java.base/share/classes/java/util/Map.java#L1348) and the `orElse` variant is easier to read.)